core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java

```diff
@@ -110,15 +115,10 @@ public AbstractInvocationBuilder setSubject(Subject subject) {
 
    }
 
-   // two separate executor services, one for sync and one for async listeners
-   protected ExecutorService syncProcessor;
+   // Processor used to handle both sync and async listener notifications. The difference is that sync notifications
```

Comment:
Sure, I can mention that.

I thought about the 1 thread guaranteeing events in the proper order for async. However the problem is the user could change the thread pool and lose that behavior very easily and only having 1 thread pool, means this could get overwhelmed quite easily in the past. To be honest I would be very surprised if anyone utilized this. And we never documented this before, we just stated it was ran asynchronously.